src/app/screens/sendStx/index.tsx

```diff
@@ -1,162 +1,143 @@
-  let amountToSend: string | undefined;
-  let stxMemo: string | undefined;
+import styled from 'styled-components';
+import { Step, getNextStep, getPreviousStep } from './stepResolver';
+import Step1SelectRecipientAndMemo from './steps/Step1SelectRecipient';
+import Step2SelectAmount from './steps/Step2SelectAmount';
+import Step3Confirm from './steps/Step3Confirm';
 
-  if (location.state) {
-    recipientAddress = location.state.recipientAddress;
-    amountToSend = location.state.amountToSend;
-    stxMemo = location.state.stxMemo;
-  }
+const Container = styled.div`
+  display: flex;
+  flex: 1 1 100%;
+  min-height: 370px;
+`;
 
-  const { isLoading, data, mutate } = useMutation<
-    StacksTransaction,
-    Error,
-    { associatedAddress: string; amount: string; memo?: string }
-  >({
-    mutationFn: async ({ associatedAddress, amount, memo }) => {
-      const unsignedSendStxTx: StacksTransaction =
-        await generateUnsignedStxTokenTransferTransaction(
-          associatedAddress,
-          stxToMicrostacks(new BigNumber(amount)).toString(),
-          memo!,
-          stxPendingTxData?.pendingTransactions ?? [],
-          stxPublicKey,
-          selectedNetwork,
-        );
-      applyFeeMultiplier(unsignedSendStxTx, feeMultipliers);
-      return unsignedSendStxTx;
-    },
-  });
+const TitleContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  flex: 0 0;
+`;
 
-  useEffect(() => {
-    if (data) {
-      navigate('/confirm-stx-tx', {
-        state: {
-          unsignedTx: buf2hex(data.serialize()),
-        },
-      });
-    }
-  }, [data]);
+const Title = styled.div`
+  ${(props) => props.theme.typography.headline_xs}
+  margin-top: ${(props) => props.theme.spacing(6)}px;
+  margin-bottom: ${(props) => props.theme.spacing(12)}px;
+`;
 
-  const handleBackButtonClick = () => {
-    // redirect to homepage to avoid looping back to confrim screen
-    navigate('/');
-  };
-
-  function validateFields(associatedAddress: string, amount: string, memo: string): boolean {
-    if (!associatedAddress) {
-      setAddressError(t('ERRORS.ADDRESS_REQUIRED'));
-      return false;
-    }
+function SendStxScreen() {
+  const isInOption = isInOptions();
+  const navigate = useNavigate();
+  const { t } = useTranslation('translation');
 
-    if (!amount) {
-      setAmountError(t('ERRORS.AMOUNT_REQUIRED'));
-      return false;
-    }
-    if (!validateStxAddress({ stxAddress: associatedAddress, network: network.type })) {
-      setAddressError(t('ERRORS.ADDRESS_INVALID'));
-      return false;
-    }
+  const location = useLocation();
+  const {
+    recipientAddress: stateAddress,
+    amountToSend,
+    stxMemo,
+    fee: previousFee,
+  } = location.state || {};
 
-    if (associatedAddress === stxAddress) {
-      setAddressError(t('ERRORS.SEND_TO_SELF'));
-      return false;
-    }
+  const [currentStep, setCurrentStep] = useState<Step>(stateAddress && amountToSend ? 1 : 0);
```

Comment:
Would be better to use the enum values rather than 0 and 1.

Also, shouldn't we go to the amount selection screen if the address is set, regardless of whether an amount is set? 🤔

Reply:
Should that not maybe be if address and the memo have values instead of the address and amount?